network/network.go

```diff
@@ -79,6 +82,29 @@ type Network struct {
 	storeProvider       storage.Provider
 }
 
+// CheckHealth performs health checks for the network engine.
+func (n *Network) CheckHealth(_ context.Context) map[string]core.HealthCheckResult {
+	results := make(map[string]core.HealthCheckResult)
+	if n.certificate.Leaf != nil {
+		// TLS enabled, verify the configured certificate
```

Comment:
Bit weird of a health check. Seems more like something for fail-fast, no?

Reply:
See discussion above